providers/cloudflare/rest.go

```diff
@@ -505,6 +507,83 @@ func (c *cloudflareProvider) sendPageRule(endpoint, method string, data string)
 	return err
 }
 
+func (c *cloudflareProvider) getWorkerRoutes(id string, domain string) ([]*models.RecordConfig, error) {
+	url := fmt.Sprintf(workerRoutesURL, id)
+	data := workerRouteResponse{}
+	if err := c.get(url, &data); err != nil {
+		return nil, fmt.Errorf("failed fetching worker route list from cloudflare: %s", err)
+	}
+	if !data.Success {
+		return nil, fmt.Errorf("failed fetching worker route list cloudflare: %s", stringifyErrors(data.Errors))
+	}
+	recs := []*models.RecordConfig{}
+	for _, pr := range data.Result {
+		var thisPr = pr
+		r := &models.RecordConfig{
+			Type:     "WORKER_ROUTE",
+			Original: thisPr,
+			TTL:      1,
+		}
+		r.SetLabel("@", domain)
+		r.SetTarget(fmt.Sprintf("%s,%s", // $PATTERN,$SCRIPT
+			pr.Pattern,
+			pr.Script))
+		recs = append(recs, r)
+	}
+	return recs, nil
+}
+
+func (c *cloudflareProvider) deleteWorkerRoute(recordID, domainID string) error {
+	endpoint := fmt.Sprintf(singleWorkerRouteURL, domainID, recordID)
+	req, err := http.NewRequest("DELETE", endpoint, nil)
+	if err != nil {
+		return err
+	}
+	c.setHeaders(req)
+	_, err = handleActionResponse(http.DefaultClient.Do(req))
+	return err
+}
+
+func (c *cloudflareProvider) updateWorkerRoute(recordID, domainID string, target string) error {
+	if err := c.deleteWorkerRoute(recordID, domainID); err != nil {
+		return err
+	}
+	return c.createWorkerRoute(domainID, target)
+}
+
+func (c *cloudflareProvider) createWorkerRoute(domainID string, target string) error {
+	endpoint := fmt.Sprintf(workerRoutesURL, domainID)
+	return c.sendWorkerRoute(endpoint, "POST", target)
+}
+
+func (c *cloudflareProvider) sendWorkerRoute(endpoint, method string, data string) error {
+	// $PATTERN,$SCRIPT
+	parts := strings.Split(data, ",")
+	if len(parts) != 2 {
+		return fmt.Errorf("unexpected data: '%s' (expected: 'PATTERN,SCRIPT')", data)
+	}
+	pattern := parts[0]
+	script := parts[1]
+	wr := &workerRoute{
+		Pattern: pattern,
+		Script:  script}
```

Comment:
Please format the last item like this so that diff's in the future will be smaller.

```
    Script: script,
}
```